piecrust/src/session.rs

```diff
@@ -554,11 +555,11 @@ impl Session {
 
         let mut serializer =
             StandardBufSerializer::new(ser, scratch, Infallible);
-        serializer.serialize_value(value).expect("Infallible");
+        serializer.serialize_value(value)?;
 
         let pos = serializer.pos();
 
-        buf[..pos].to_vec()
+        Ok(buf[..pos].to_vec())
```

Comment:
Ditto

Reply:
If the type is too large to fit in `MAX_META_SIZE` then this will error out anyway. Fortunately